File: lib/test/structuredfield/parseItem.test.ts
import { parseItem } from '@svta/common-media-library/structuredfield/parse/parseItem';
import { SfItem } from '@svta/common-media-library/structuredfield/SfItem';
import assert from 'node:assert';
import test from 'node:test';

test('test parseItem', () => {
	assert.deepStrictEqual(parseItem(`123;a=1;b`), { value: new SfItem(123, { 'a': 1, 'b': true }), src: `` });
	assert.throws(() => parseItem(``), /failed to parse "" as Token/);
});

File: python/src/test/structuredfield/parseItem_test.py
import unittest

from ...structuredfield.parse.ParsedValue import ParsedValue
from ...structuredfield.parse.parseItem import parseItem
from ...structuredfield.SfException import SfException
from ...structuredfield.SfItem import SfItem


class TestDecodeSfItem(unittest.TestCase):
    def test_parseItem(self):
        self.assertEqual(
            parseItem("123;a=1;b"), ParsedValue(SfItem(123, {"a": 1, "b": True}), "")
        )
        self.assertRaises(SfException, parseItem, "")
